src/cli/opentelemetry.rs

```diff
@@ -0,0 +1,65 @@
+use eyre::Result as EyreResult;
+use opentelemetry::{
+    sdk::{
+        trace::{self, IdGenerator, Sampler},
+        Resource,
+    },
+    KeyValue,
+};
+use opentelemetry_http::{HeaderExtractor, HeaderInjector};
+use opentelemetry_otlp::WithExportConfig;
+use std::collections::HashMap;
+use structopt::StructOpt;
+use tracing::{info, Subscriber};
+use tracing_subscriber::{registry::LookupSpan, Layer};
+
+#[derive(Clone, Debug, PartialEq, StructOpt)]
+pub struct Options {
+    /// OpenTelemetry submission
+    #[structopt(long)]
+    pub opentelemetry: bool,
+}
+
+impl Options {
+    pub fn to_layer<S>(&self) -> EyreResult<impl Layer<S>>
+    where
+        S: Subscriber + for<'span> LookupSpan<'span>,
+    {
+        if !self.opentelemetry {
+            return Ok(None);
+        }
+
+        let addr = std::net::SocketAddr::from(([127, 0, 0, 1], 6831));
+
+        let mut map = HashMap::new();
+        // TODO replace <TOKEN> with the authentication token created in section 2 above
```

Comment:
Does this need to be resolved before this PR is merged? Or is this a longer term TODO?
Also what is section 2 exactly?